Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -25,6 +25,34 @@ namespace AzToolsFramework
 {
     namespace Prefab
     {
+        namespace Internal
+        {
+            struct PrefabPublicNotificationBusHandler final
```

Comment:
To keep the PrefabSystemComponent code compact and maintainable, I recommend that you make a m_PrefabPublicNotificationsHandler similar to m_prefabPublicRequestHandler and move the logic and the reflect methods into its own header and cpp files.

Reply:
Will fix this.

Reply:
I checked out all the code related to `AZ_EBUS_BEHAVIOR_BINDER` and it seems that it is always added into `*SystemComponent.cpp code` directly without making it as a member variable of any system component or having its own reflect function inside the handler class. So I will keep the code this way.

Reply:
Other systems behavior shouldn't define how we organize code in this component. If you run into any issues moving this code to its own files, then we can consider keeping it as is. But I recommend that you give it a try first.